packages/visualizations/src/components/MapPoi/Map.ts

```diff
@@ -59,6 +168,10 @@ export default class MapPOI {
      * - `baseStyle` could be removed
      * - The key block could also be removed in MapRender.svelte
      */
+    /**
+     * Update the sources and layers of the map.
+     * Layers of the map base style are untouched.
+     */
```

Comment:
I guess this function should clean the opened popups and popupFeatures array as they are associated to a specific source that can change here. I have had a bug while editing a map when I have clicked to open a tooltip, leave it opened and then changed the source.

Adding : 
```
this.popup.remove();
this.popupFeatures = [];
```

Do the trick for my case but probably we will want something cleaner at some point and more general, what are the other things to clean when changing the source ? layerIds ?

Reply:
Should be good now with https://github.com/opendatasoft/ods-dataviz-sdk/pull/186/commits/8a46de55acbbba197f4482669c3e8f8ae6c6bf15

Please test it with a local version of the SDK

Reply:
Thanks ! It seems to work as expected now on source changes, but I still have the same error when removing the map instance (for example, you have a configured map, open a tooltip, and then change the block type while the tooltip is open) It seems that the popup is not removed in the map.remove function ! 

Adding line 170 this code fixes it: 
```
remove() {
        this.popup.remove();
        this.popupFeatures = [];
        this.queue((map) => map.remove());
    } 
```

Reply:
ah yes, I didn't test when the map is destroyed, thanks !